apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/util/InstanceJsonPropertiesUtil.java

```diff
@@ -45,7 +55,32 @@ public static List<KeyStringValuePair> parseProperties() {
 
         properties.add(KeyStringValuePair.newBuilder().setKey("namespace").setValue(Config.Agent.NAMESPACE).build());
         properties.add(KeyStringValuePair.newBuilder().setKey("cluster").setValue(Config.Agent.CLUSTER).build());
+        properties.add(KeyStringValuePair.newBuilder().setKey("version").setValue(getAgentVersion()).build());
 
         return properties;
     }
+
+    public static String getAgentVersion() {
+        try {
+            InputStream inStream = InstanceJsonPropertiesUtil.class.getClassLoader()
+                    .getResourceAsStream(GIT_PROPERTIES);
+            if (inStream != null) {
+                Properties gitProperties = new Properties();
+                gitProperties.load(inStream);
+                String commitIdAbbrev = gitProperties.getProperty("git.commit.id.abbrev");
+                String buildTime = gitProperties.getProperty("git.build.time");
+                String buildVersion = gitProperties.getProperty("git.build.version");
+                String version =  buildVersion + "-" + commitIdAbbrev + "-" + buildTime;
+
+                LOGGER.info("SkyWalking agent version: {}", version);
+                return version;
+            } else {
+                LOGGER.warn("{} not found, SkyWalking agent version: unknown", GIT_PROPERTIES);
```

Comment:
@kezhenxu94 If users build from the source tarball, here is the WARN level logging.